api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -68,6 +68,19 @@ message TcpProxy {
 
     // The hostname to send in the synthesized CONNECT headers to the upstream proxy.
     string hostname = 1 [(validate.rules).string = {min_len: 1}];
+
+    // Use POST method instead of CONNECT method to tunnel the TCP stream.
+    // The 'protocol: bytestream' header is also NOT set to comply with the HTTP spec.
+    // Only HTTP/2 is supported.
+    //
+    // The upstream proxy is expected to convert POST back to CONNECT.
+    bool use_post = 2;
+
+    // Additional request headers to upstream proxy. This is mainly used to
+    // trigger upstream to convert POST requests back to CONNECT requests.
+    // The hostname header can be overridden by this field.
```

Comment:
@envoyproxy/api-shepherds thoughts on this?
Given we have the explicit hostname field I think we should disallow.  I'm inclined to say we disallow Host: overall given the  HTTP/1 and HTTP/2 weirdness.  WDYT of doing this globally, stating here that it's unsupported but unenforced and adding a TODO for me to enforce?

Reply:
Sounds good, I will wait for https://github.com/envoyproxy/envoy/pull/14747 to merge in

Reply:
@alyssawilk yeah, I tend to agree. We already had one PR with `ext_proc` that we decided to disallow `host` override via header mutation.

Reply:
Tagging @gbrail and https://github.com/envoyproxy/envoy/pull/14385.

Reply:
Thanks, added a config_test.